editor.planx.uk/src/pages/FlowEditor/lib/store/editor.ts

```diff
@@ -487,4 +492,29 @@ export const editorStore: StateCreator<
     externalPortals[id] = { name, href };
     set({ externalPortals });
   },
+
+  getURLForNode: (nodeId) => {
+    const { orderedFlow: flow, flowSlug, teamSlug } = get();
+    if (!flow) throw Error("Missing ordered flow!");
+
+    const path = getPathForNode({ nodeId, flow });
+    const internalPortals = path.filter(
+      ({ type }) => type === ComponentType.InternalPortal,
+    );
+    const [node, parent, grandparent] = path;
+
+    // Construct the internal portal path if applicable
+    const portalPath = internalPortals.length
+      ? "," + internalPortals.map(({ id }) => id).join(",")
+      : "";
+
+    // Determine node path based on the node type
+    const nodePath =
+      node.type === ComponentType.Answer
+        ? `nodes/${grandparent.id}/nodes/${parent.id}/edit`
+        : `nodes/${parent.id}/nodes/${node.id}/edit`;
```

Comment:
I _think_ this is the only variance that controls the path used for the modal - please let me know if there's something I've missed here!

Reply:
This seems right to me here! Only edge case I possibly considered here were "Filters" - but these links work as expected testing on pizza :raised_hands: 

Plus very cool fun bonus - Search now makes visible the (wild!) fact that Filter's have a default data field `flag` assigned to them via the component's `handleSubmit` and later relied on in `upcomingCardIds()`: 
![Screenshot from 2024-09-11 21-35-30](https://github.com/user-attachments/assets/e8702c2a-d6c2-4b16-a126-0ec95bcd831f)

Reply:
Ha! Good catch and a fun little easter egg... 🙃